### PR TITLE
deprecate albIngress field and replace with awsLoadBalancerController

### PR DESCRIPTION
**Pull request by aclevername**
### Description

Closes https://github.com/weaveworks/eksctl/issues/4675

Test:

Both set in same ng:
```
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: jk
  region: us-west-2

nodeGroups:
  - name: invalid-ng
    iam:
      withAddonPolicies:
        awsLoadBalancerController: true
        albIngress: true
````

```
eksctl create nodegroup -f cluster.yaml
Error: couldn't create cluster provider from options: "awsLoadBalancerController" and "albIngress" cannot both be configured,please use "awsLoadBalancerController" as "albIngress" is deprecated
```

Valid nodegroups with old and new field:
```
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: jk
  region: us-west-2

nodeGroups:
  - name: alb-ng
    iam:
      withAddonPolicies:
        albIngress: true
  - name: aws-controller-ng
    iam:
      withAddonPolicies:
        awsLoadBalancerController: true
```

```
eksctl create nodegroup -f cluster.yaml
2022-02-03 10:47:54 [!]  nodegroup.iam.WithAddonPolicies.albIngress field is deprecated, please us awsLoadBalancerController instead
...
2022-02-03 10:48:08 [ℹ]  waiting for CloudFormation stack "eksctl-jk-nodegroup-aws-controller-ng"
2022-02-03 10:48:09 [ℹ]  waiting for CloudFormation stack "eksctl-jk-nodegroup-alb-ng"
...
2022-02-03 10:53:08 [✔]  created 2 nodegroup(s) in cluster "jk"
2022-02-03 10:53:08 [✔]  created 0 managed nodegroup(s) in cluster "jk"
2022-02-03 10:53:11 [ℹ]  checking security group configuration for all nodegroups
2022-02-03 10:53:11 [ℹ]  all nodegroups have up-to-date cloudformation templates
```

both nodes contain the correct IAM template for the loadbalancer controller

### CODE REVIEW

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 1]
```diff
@@ -755,6 +761,18 @@ func validateNodeGroupIAMWithAddonPolicies(
 	return nil
 }
 
+func validateDeprecatedIAMFields(iam *NodeGroupIAM) error {
+	if IsEnabled(iam.WithAddonPolicies.DeprecatedALBIngress) {
+		if IsEnabled(iam.WithAddonPolicies.AWSLoadBalancerController) {
+			return fmt.Errorf(`"awsLoadBalancerController" and "albIngress" cannot both be configured, ` +
+				`please use "awsLoadBalancerController" as "albIngress" is deprecated`)
+		}
+		logger.Warning("nodegroup.iam.WithAddonPolicies.albIngress field is deprecated, please us awsLoadBalancerController instead")
```
Comment (nikimanoledaki) [inline pkg/apis/eksctl.io/v1alpha5/validation.go:770]:
```suggestion
		logger.Warning("nodegroup.iam.WithAddonPolicies.albIngress field is deprecated, please use awsLoadBalancerController instead")
```

File: pkg/cfn/builder/iam_helper.go [hunk 2]
```diff
@@ -138,7 +138,7 @@ func createRole(cfnTemplate cfnTemplate, clusterIAMConfig *api.ClusterIAM, iamCo
 		cfnTemplate.attachAllowPolicy("PolicyEFSEC2", refIR, efsEc2Statements())
 	}
 
-	if api.IsEnabled(iamConfig.WithAddonPolicies.AWSLoadBalancerController) {
+	if api.IsEnabled(iamConfig.WithAddonPolicies.AWSLoadBalancerController) || api.IsEnabled(iamConfig.WithAddonPolicies.DeprecatedALBIngress) {
```
Comment (nikimanoledaki) [inline pkg/cfn/builder/iam_helper.go:141]: it adds `PolicyAWSLoadBalancerController` even if `ALBIngress` is set? 🤔 shouldn't it error out and ask to use `AWSLoadBalancerController` explicitly instead?

Comment (aclevername) [reply pkg/cfn/builder/iam_helper.go:141]: we wanna support both, but log a warning if the user is using `ALBIngress`. Eventually when we remove `ALBIngress` we will want it to error

**Review by nikimanoledaki — APPROVED** [review]

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 3]
```diff
@@ -755,6 +761,18 @@ func validateNodeGroupIAMWithAddonPolicies(
 	return nil
 }
 
+func validateDeprecatedIAMFields(iam *NodeGroupIAM) error {
+	if IsEnabled(iam.WithAddonPolicies.DeprecatedALBIngress) {
+		if IsEnabled(iam.WithAddonPolicies.AWSLoadBalancerController) {
+			return fmt.Errorf(`"awsLoadBalancerController" and "albIngress" cannot both be configured, ` +
+				`please use "awsLoadBalancerController" as "albIngress" is deprecated`)
+		}
+		logger.Warning("nodegroup.iam.WithAddonPolicies.albIngress field is deprecated, please use awsLoadBalancerController instead")
```
Comment (cPu1) [inline pkg/apis/eksctl.io/v1alpha5/validation.go:770]: Minor nit:
```suggestion
		logger.Warning("nodegroup.iam.withAddonPolicies.albIngress field is deprecated, please use awsLoadBalancerController instead")
```

**Review by cPu1 — APPROVED** [review]
LGTM.

**Pull request merged**